### PR TITLE
Protect keyspace stat operations with mutex

### PR DESCRIPTION
**Pull request by VipinRaiP**
### Summary

Support for keyspace stats at store level and protect operations using mutex. 

### Changes

- Introduce store level struct for keyspace stats.
- Protect keyspace operations with mutex.
- Unit tests.

Issue : #296

### CODE REVIEW

File: core/stats.go [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
-var KeyspaceStat [4]map[string]int
+import (
+	"sync"
+)
 
-func UpdateDBStat(num int, metric string, value int) {
-	KeyspaceStat[num][metric] = value
+type KeyspaceStat struct {
+	stats map[string]int
+	mutex sync.Mutex
```
Comment (JyotinderSingh) [inline core/stats.go:9]: @soumya-codes do you think its okay for us to use this in-built lock for managing the KeyspaceStat? Other mutexes use the functional pattern defined in the locks.go file.

I don't have strong opinions on this. The proposed method in this PR looks good and concise for the use case, however, it deviates from the existing pattern.

Comment (soumya-codes) [reply core/stats.go:9]: IMO it is good to follow the same pattern across the code. 
Also, In this case, it appears we will not take any additional performance hit by adding a lock over the whole operation like we do with other lock types.

File: core/store.go [hunk 2]
```diff
@@ -256,9 +257,7 @@ func (store *Store) Rename(sourceKey, destKey string) bool {
 		// Remove the source key
 		delete(store.store, sourcePtr)
 		delete(store.keypool, sourceKey)
-		if KeyspaceStat[0] != nil {
-			KeyspaceStat[0]["keys"]--
-		}
+		store.keyspaceStat.IncrStat("keys")
```
Comment (soumya-codes) [inline core/store.go:260]: Why are we incrementing the key count on rename?

**Comment by JyotinderSingh** [comment]
Please rebase the PR, we just had a large change go in which might've introduced conflicts.

**Comment by JyotinderSingh** [comment]
We have removed store locks from the codebase, we no longer need this change. But thanks a lot for the effort!

**Pull request closed without merging**